### ISSUE TITLE
gene symbol lookup (search) works for hg19, sometimes not hg38

### ISSUE THREAD

**Issue by paul-shannon** [igvteam/igv.js#952]
@jrobinso 
Confirmed with today's snapshot - but I am not completely sure this won't turn out to be my error:

Entering some gene symbols in the search box ("FOXO6" is my test case) leads to "Not found" with hg38, but works as expected with hg19.

**Comment by jrobinso** [comment]
Ahh, its a problem with our database, arguably.  The FOXO6 gene has been
placed in 2 locations in the hg38 refseg file.  One of those locations
is chr1_KN196472v1_fix.    This is not a chromosome IGV knows.   I can
probably fix this by updating the database to remove entries on strange
chromosomes.   I think there are probably many fewer of these for hg19.

**Comment by paul-shannon** [comment]
Thanks, Jim!

Though there may be costs to this I have not anticipated, I vote for including only the canonical chromosomes in the reference genomes you kindly provide.

 - Paul

**Comment by jrobinso** [comment]
Yes, that's the only thing that makes sense.    Although for most
assemblies we also provide all the scraps and contigs.   Ultimately we
should present all locations to the user and let them choose, but having a
unique location is really convenient and its been like that for 10 years
with no complaints.

We just switched to an Amazon "dynamoDB" no-sql database for this and I
think we introduced these weird locations then.  I'm changing our update
script to account for this.  In the case of multiple entries the following
rules will apply

(1) keep entry with shortest chromosome name
(2) in case of a tie,  keep entry of longest feature

The first case should eliminate the contigs when a canonical name is
available.   The second will select the largest of the isoforms.    I'm
working on this now, but traveling tomorrow so it might be Friday before
the database is updated.
